tests/x11/firefox_nss.pm

```diff
@@ -0,0 +1,88 @@
+# SUSE's openQA tests - FIPS tests
+#
+# Copyright © 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Case #1560076 - FIPS: Firefox Mozilla NSS
+
+use base "x11test";
+use strict;
+use testapi;
+
+sub quit_firefox {
+    send_key "alt-f4";
+    if (check_screen("firefox-save-and-quit", 10)) {
+        send_key "ret";
+    }
+}
+
+sub run() {
+    # define fips password for firefox, and it should be consisted by:
+    # - at least 8 characters
+    # - at least one upper case
+    # - at least one non-alphabet-non-number character (like: @-.=%)
+    my $fips_password = 'openqa@SUSE';
+    ensure_installed('mozilla-nss-tools');
+
+    # launch firefox first to generate its default profile
+    x11_start_program("firefox");
+    assert_screen("firefox-launch", 90);    # launch firefox need time
+    quit_firefox;
+
+    # enable fips mode for mozilla firefox
+    x11_start_program("xterm");
+    validate_script_output 'echo | modutil -dbdir ~/.mozilla/firefox/*.default -fips true', sub { m/FIPS mode enabled/ };
```

Comment:
Why do you enable FIPS mode using this mechanism instead of the one documented by Firefox? https://support.mozilla.org/en-US/kb/Configuring%20Firefox%20for%20FIPS%20140-2